content/chainguard/chainguard-images/tooling/building-java-containers-with-jib.md

```diff
@@ -0,0 +1,588 @@
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <from>
+            <image>cgr.dev/chainguard/jre:latest</image>
+            <platforms>
+              <platform>
+                <os>linux</os>
+                <architecture>amd64</architecture>
+              </platform>
+              <platform>
+                <os>linux</os>
+                <architecture>arm64</architecture>
+              </platform>
+            </platforms>
+          </from>
+          <to>
+            <image>linky</image>
+          </to>
+          <container>
+            <ports>
+              <port>8080</port>
+            </ports>
+            <creationTime>USE_CURRENT_TIMESTAMP</creationTime>
+          </container>
+        </configuration>
+      </plugin>
+```
+
+
+The configuration specifies several settings:
+
+  - **Build goal**: The execution goal is defined as `dockerBuild`. This will build a container image for the application and then load the built image into the local Docker instance, allowing you to immediately run and test the container. 
+  - **Base (or "from") image**: Uses Chainguard's JRE image (`cgr.dev/chainguard/jre:latest`). This example explicitly specifies both `arm64` and `amd64` architectures. If you don't do this, Jib will build an `amd64` image only, regardless of host architecture. 
+  - **Target image**: Name of the image to build (`linky`). The tag will default to `latest` if not specified.
+  - **Port**: Documents that port `8080` is used for the Spring Boot application
+  - **Creation time**: Uses the current timestamp
+
+
+## Stage 3 — Building Container Images with Jib
+
+With Jib configured, you can now build your containerized Java application.
+Because we have specified `dockerBuild` as the execution goal, this step will
+require a local Docker install. If you want to avoid using Docker and push the
+image straight to a registry, jump to Stage 5.
+
+Run:
+
+```shell
+mvn clean install
+```
+
+You’ll get output like the following:
+
+```
+...
+[WARNING] Detected multi-platform configuration, only building image that matches the local Docker Engine's os and architecture (linux/arm64) or the first platform specified
+[INFO]
+[INFO]
+[INFO] Container entrypoint set to [java, -cp, @/app/jib-classpath-file, com.example.demo.DemoApplication]
+[INFO] Container entrypoint set to [java, -cp, @/app/jib-classpath-file, com.example.demo.DemoApplication]
+[INFO]
+[INFO] Built image to Docker daemon as linky
+...
+```
+
+
+This command has compiled the application, built the container, and loaded it
+into the local Docker instance.
+
+To verify that the image was built:
+
+```shell
+docker images linky
+```
+
+Which will provide details on the new image:
+
+```
+REPOSITORY   TAG       IMAGE ID       CREATED         SIZE
+linky        latest    43171844f68e   3 minutes ago   465MB
+```
+
+You should also scan the image for CVEs. This example uses
```

Comment:
I feel like this step should be after building the image, it seems a little premature / odd here?

Reply:
I updated the versions in the pom.xml and there's no vulns now.